### PR TITLE
Feature: Add "image" column

### PR DESCRIPTION
**Pull request by danilopolani**
Add a `Image` column type to tables.

> **Notes for mainters**: I used type hinting whenever possible although they were not used in the rest of the code. Please feel free to review the codestyle about that and about one line "if" :)
> An assets regeneration **is needed** to have the `bg-cover` class from Tailwind.

**Usage**:

```php
Image::make($name)
    ->width($value) // Set the image width. Can be both a number (used as "pixel") or a string, like "2rem".
    ->height($value) // Set the image height. Can be both a number (used as "pixel") or a string, like "2rem".
    ->square($value) // Shortcut to set both width and height to the given value.
    ->class($class) // Additional classes for the image.
    ->rounded() // Show the image with rounded borders.
    ->cover(); // Resize the background image to cover the entire container. Useful to display non-squared images as squares.
```

**Example**:

```php
Image::make('picture')
    ->square(64)
    ->rounded()
    ->cover();
```
Output:

image

### CODE REVIEW

File: packages/tables/resources/views/cells/image.blade.php [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    target="_blank"
+    @if ($column->cover)
+        class="block bg-center {{ $column->getClass() }}"
+        style="
+            {{ $column->getWidth() ? "width: {$column->getWidth()};" : '' }}
+            {{ $column->getHeight() ? "height: {$column->getHeight()};" : '' }}
+            background-image: url({{ $column->getValue($record) }});"
+    @endif
+>
+    @if (!$column->cover)
```
Comment (ryangjchandler) [inline packages/tables/resources/views/cells/image.blade.php:12]: Not sure what @danharrin thinks, but `@unless` is being used elsewhere in the code base. Might be good for consistency.

**Review by ryangjchandler — CHANGES_REQUESTED** [review]

File: packages/tables/resources/views/cells/image.blade.php (hunk 1 above)
Comment (danilopolani) [reply packages/tables/resources/views/cells/image.blade.php]: Right, changed!

**Comment by walli55** [comment]
Alomaisi

**Comment by walli55** [comment]
[]() of

**Comment by danharrin** [comment]
Thanks! I ended up removing the `cover()` option and setting a default height. I also added an option to set the disk that the files should be read from, similar to `FileUpload`.

**Comment by danilopolani** [comment]
> Thanks! I ended up removing the `cover()` option and setting a default height. I also added an option to set the disk that the files should be read from, similar to `FileUpload`.
> 
> The `FileUpload` component has an option to autocrop to `1:1` aspect ratio if you're looking for square image uploads 👍

Thank you @danharrin !

I'm not 100% sure about the disk: can I bypass it? I.e. can I show an image from an external URL?

**Comment by danharrin** [comment]
> > Thanks! I ended up removing the `cover()` option and setting a default height. I also added an option to set the disk that the files should be read from, similar to `FileUpload`.
> > The `FileUpload` component has an option to autocrop to `1:1` aspect ratio if you're looking for square image uploads 👍
> 
> Thank you @danharrin !
> 
> I'm not 100% sure about the disk: can I bypass it? I.e. can I show an image from an external URL?

It uses `Storage::url()` under the hood which will respect external URLs if they are configured on your disk correctly - i.e. s3 URLs that are already fully qualified when they are saved. As for external URLs that reside outside of the storage disk, maybe we should check for a valid URL before passing through `Storage::url()`?

**Comment by danilopolani** [comment]
> It uses `Storage::url()` under the hood which will respect external URLs if they are configured on your disk correctly - i.e. s3 URLs that are already fully qualified when they are saved. As for external URLs that reside outside of the storage disk, maybe we should check for a valid URL before passing through `Storage::url()`?

I think we should do the check, I just tried running `Storage::url('https://www.google.com/images/branding/googlelogo/1x/googlelogo_color_150x54dp.png')` and the result is `"/storage/https://www.google.com/images/branding/googlelogo/1x/googlelogo_color_150x54dp.png"` 🤔 

Maybe `filter_var` is enough for the check:

```php
var_dump(filter_var('https://www.google.com/images/branding/googlelogo/1x/googlelogo_color_150x54dp.png', FILTER_VALIDATE_URL)); // string
var_dump(filter_var('/images/branding/googlelogo/1x/googlelogo_color_150x54dp.png', FILTER_VALIDATE_URL)); // false
```

**Comment by danharrin** [comment]
#117

**Pull request merged**